web_src/src/pages/canvas/components/nodes/handle.tsx

```diff
@@ -194,12 +194,14 @@ function TooltipContent({ connections = [], conditions = [] }: { connections?: S
   }
 
   function ApprovalContent({ approval }: { approval: SuperplaneConditionApproval }) {
+    const totalCount = approval.from?.reduce((sum, req) => sum + (req.count || 1), 0) || 0;
```

Comment:
This semi-complex logic appears two times in the code, here and for the settings tab. A shared function that encapsulates this logic would help. e.g. `requiredApprovals(approval)`.